### PR TITLE
Hide Kube Java lib invocations behind internal API

### PR DESCRIPTION
**Pull request by tbak**
Hide Kube Java lib invocations behind internal API. We need a proper interface between Kube and Titus so we can easily inject stubbed version of Kube Cluster in the integration tests.

### CODE REVIEW

File: titus-common-runtime/src/main/java/com/netflix/titus/runtime/connector/kubernetes/KubeApiFacade.java [hunk 1]
```diff
@@ -18,26 +18,49 @@
...
+ */
 public interface KubeApiFacade {
 
-    ApiClient getApiClient();
-
-    CoreV1Api getCoreV1Api();
+    // Nodes
 
-    CustomObjectsApi getCustomObjectsApi();
+    void deleteNode(String nodeName) throws ApiException;
```
Comment (corindwyer) [inline titus-common-runtime/src/main/java/com/netflix/titus/runtime/connector/kubernetes/KubeApiFacade.java:36]: Maybe we shouldn't leak the underlying client's ApiException? Thoughts?

**Review by corindwyer — APPROVED** [review]

File: titus-common-runtime/src/main/java/com/netflix/titus/runtime/connector/kubernetes/KubeApiFacade.java (hunk 1 above)
Comment (tbak) [reply titus-common-runtime/src/main/java/com/netflix/titus/runtime/connector/kubernetes/KubeApiFacade.java:36]: Good idea. I will create a wrapper

**Review by corindwyer — APPROVED** [review]

File: titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/controller/PersistentVolumeClaimGcController.java [hunk 2]
```diff
@@ -107,20 +106,11 @@ private boolean gcPersistentVolumeClaim(V1PersistentVolumeClaim pvc) {
...
-                    null,
-                    null
-            );
+            kubeApiFacade.deleteNamespacedPersistentVolumeClaim(DEFAULT_NAMESPACE, volumeClaimName);
             logger.info("Successfully deleted persistent volume claim {}", formatPvcEssentials(pvc));
             return true;
-        } catch (ApiException e) {
-            if (!e.getMessage().equalsIgnoreCase(NOT_FOUND)) {
+        } catch (KubeApiException e) {
+            if (e.getErrorCode() != KubeApiException.ErrorCode.NOT_FOUND) {
```
Comment (fabiokung) [inline titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/controller/PersistentVolumeClaimGcController.java:113]: if I read this code correctly, shouldn't this be an `==` comparison? If I'm right, this bug was already present before your change.

File: titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/controller/PersistentVolumeUnassociatedGcController.java [hunk 3]
```diff
@@ -113,19 +112,11 @@ private boolean gcPersistentVolume(V1PersistentVolume pv) {
...
-                    null,
-                    null
-            );
+            kubeApiFacade.deletePersistentVolume(volumeName);
             logger.info("Successfully deleted persistent volume {}", formatPvEssentials(pv));
             return true;
-        } catch (ApiException e) {
-            if (!e.getMessage().equalsIgnoreCase(NOT_FOUND)) {
+        } catch (KubeApiException e) {
+            if (e.getErrorCode() != KubeApiException.ErrorCode.NOT_FOUND) {
```
Comment (fabiokung) [inline titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/controller/PersistentVolumeUnassociatedGcController.java:119]: same from before, this looks like it should be an `==`

**Review by fabiokung — APPROVED** [review]

File: titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/controller/PersistentVolumeClaimGcController.java (hunk 2 above)
Comment (tbak) [reply titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/controller/PersistentVolumeClaimGcController.java:113]: Good catch. I kept the original logic which was also incorrect.

**Pull request merged**